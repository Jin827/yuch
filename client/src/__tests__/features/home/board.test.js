import React from 'react';
import { shallow, mount } from 'enzyme';
import Board from '../../../features/home/board';

describe('<Board />', () => {
  const wrapper = shallow(<Board />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('sets initial state', () => {
    expect(wrapper.state().lastScrollY).toBe(0);
  });

  describe('when mouted', () => {
    // let spy;
    // beforeEach(() => {

    //   wrapper.setState({ lastScrollY: 1010 });
    // });

    afterEach(() => {
      // wrapper.setState({ lastScrollY: 0 });
      // spy.mockClear();
    });

    it('should call methodName on scroll during componentDidMount', () => {
      // const map = {};
      // window.addEventListener = jest.jest.genMockFn().mockImpl((event, cb) => {
      //   map[event] = cb;
      // });
      // wrapper = mount(<Board />);
      // map.scroll({ pageY: 1010 });
      // expect(wrapper.handleScroll).toHaveBeenCalled();
      // spy = jest.spyOn(global, 'addEventListener');
      // wrapper.instance().handleScroll();
      // expect(spy).toHaveBeenCalled();
    });
    it('should update lastScrollY state', () => {});
    it('laods Map on scroll down', () => {
      // expect(wrapper.find('LoadableComponent').exists()).toBe(true);
    });
  });
});
