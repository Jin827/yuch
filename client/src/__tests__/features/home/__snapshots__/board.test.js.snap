// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> should render successfully 1`] = `
<Board>
  <div
    className="homeboard"
    id="board"
  >
    <div
      className="homeboard--text"
    >
      <div
        className="mb4"
      >
        <h3
          className="tc"
        >
          - Hours -
        </h3>
        <div
          className="flex justify-center"
        >
          <p
            className="mr3 float-left"
          >
            평일
            <br />
            <br />
            공휴일/주말
          </p>
          <p>
            중식  11 : 30 - 13 : 30
            <br />
            석식  16 : 30 - 18 : 00
            <br />
            중식  11 : 30 - 13 : 30
          </p>
        </div>
      </div>
      <div
        className="tc"
      >
        <h3>
          - Contact -
        </h3>
        <p>
          경주시 황성동 1071-1번지
          <br />
          용강공단 내 강남골프장 맞은 편
        </p>
        <br />
        <div
          className="contact-number"
        >
          <div
            className="flex justify-center pt3"
          >
            <svg
              className="mr3 mh-auto"
              height="35"
              viewBox="0 0 35 35"
              width="35"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M27.3333333,15 C26.9653333,15 26.6666667,14.7013333 26.6666667,14.3333333 C26.6666667,9.92133333 23.0786667,6.33333333 18.6666667,6.33333333 C18.2986667,6.33333333 18,6.03466667 18,5.66666667 C18,5.29866667 18.2986667,5 18.6666667,5 C23.8133333,5 28,9.188 28,14.3333333 C28,14.7013333 27.7013333,15 27.3333333,15 Z M24.3636364,15 C24.0123636,15 23.7272727,14.7149091 23.7272727,14.3636364 C23.7272727,11.556 21.444,9.27272727 18.6363636,9.27272727 C18.2850909,9.27272727 18,8.98763636 18,8.63636364 C18,8.28509091 18.2850909,8 18.6363636,8 C22.1452727,8 25,10.8547273 25,14.3636364 C25,14.7149091 24.7149091,15 24.3636364,15 Z M18,11.7142857 C18,11.32 18.32,11 18.7142857,11 C21.0771429,11 23,12.9228571 23,15.2857143 C23,15.68 22.68,16 22.2857143,16 C21.8914286,16 21.5714286,15.68 21.5714286,15.2857143 C21.5714286,13.71 20.29,12.4285714 18.7142857,12.4285714 C18.32,12.4285714 18,12.1085714 18,11.7142857 Z M26,22.1145 C26,23.8638571 23.8367143,26 21.3110714,26 C18.8220714,26 14.0286429,23.0075 12.0105714,20.9894286 C9.9925,18.9713571 7,14.1779286 7,11.6875714 C7,9.16328571 9.1375,7 10.8868571,7 L12.4896429,7 C12.7488571,7 12.985,7.14657143 13.099,7.37864286 C13.1003571,7.38 14.0652857,9.34514286 14.6977143,10.5760714 C15.3952857,11.9318571 14.0082857,13.5617857 13.2754286,14.2878571 C13.5332857,14.9705 14.1847143,16.4335 15.3762857,17.6237143 C16.5665,18.8152857 18.0308571,19.4667143 18.7121429,19.7245714 C19.4382143,18.9917143 21.0681429,17.6006429 22.4239286,18.3022857 C23.6548571,18.9347143 25.62,19.8996429 25.6213571,19.8996429 C25.8534286,20.015 26,20.2511429 26,20.509 L26,22.1145 Z M17.5,0 C8.84692308,0 0,1.83884615 0,15.4807692 C0,22.8375 2.49576923,27.0213462 8.07692308,28.9746154 L8.07692308,32.6725 C8.07692308,33.9567308 9.12019231,35 10.4044231,35 C11.0748077,35 11.7128846,34.7105769 12.1086538,34.2515385 L15.6530769,30.9292308 C16.2709615,30.9494231 16.8578846,30.9615385 17.5,30.9615385 C26.355,30.9615385 35,29.8173077 35,16.1538462 C35,1.67461538 25.0128846,0 17.5,0 Z"
                fill="#303C42"
                fillRule="evenodd"
              />
            </svg>
            <a
              aria-label="dial a phone call"
              className="td-none c-point2"
              href="tel:+82-10-8034-0057"
            >
              054 .  745  . 0999
              <br />
              010 . 8034 . 0057
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="bt tc ph f-mini f-en"
  >
    © 2018 | All right reserved | YUCH
  </footer>
</Board>
`;
